crates/zizmor/src/audit/bot_conditions.rs

```diff
@@ -200,11 +228,245 @@ impl BotConditions {
+
+            for fragment in fragments {
+                if seen_fragments.insert(fragment.clone()) {
+                    // Replace the fragment with the secure alternative
+                    let replacement = if SPOOFABLE_ACTOR_NAME_CONTEXTS.iter().any(|pattern| {
+                        // Parse the fragment to check if it matches our patterns
+                        if let Ok(test_expr) = Expr::parse(&fragment) {
+                            if let Expr::Context(ctx) = test_expr {
+                                pattern.matches(&ctx)
+                            } else {
+                                false
+                            }
+                        } else {
+                            false
+                        }
+                    }) {
+                        "github.event.pull_request.user.login"
+                    } else {
+                        // This is an actor_id context, replace with user.id
+                        "github.event.pull_request.user.id"
+                    };
+
+                    patches.push(Patch {
+                        route: step.route().with_keys(&["if".into()]),
+                        operation: Op::RewriteFragment {
+                            from: fragment.clone().into(),
+                            to: replacement.into(),
+                            after: None,
+                        },
+                    });
+                }
+            }
+
+            if !patches.is_empty() {
+                Some(Fix {
+                    title: "Replace spoofable actor context with github.event.pull_request.user.login".into(),
+                    description: "Replace spoofable actor context with github.event.pull_request.user.login to ensure the job runs as the PR author".into(),
+                    _key: step.location().key,
+                    patches,
+                })
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+
+    /// Create a fix that suggests changing from pull_request_target to pull_request
+    fn create_trigger_change_fix<'doc>(job: &impl JobExt<'doc>) -> Fix<'doc> {
+        Fix {
+            title: "Change trigger from pull_request_target to pull_request".into(),
+            description: "Change the workflow trigger from pull_request_target to pull_request to prevent running with elevated permissions".into(),
+            _key: job.location().key,
+            patches: vec![Patch {
+                route: job.parent().location().route.with_keys(&["on".into()]),
+                operation: Op::Replace({
+                    let mut map = serde_yaml::Mapping::new();
+                    map.insert(
+                        serde_yaml::Value::String("pull_request".into()),
+                        serde_yaml::Value::Mapping(Default::default()),
+                    );
+                    serde_yaml::Value::Mapping(map)
+                }),
+            }],
+        }
+    }
```

Comment:
I like the idea for this fix, but I think we should leave it out for now 😅 -- it overlaps with the fix we'll want to suggest for `dangerous-triggers`, plus it's hard to suggest confidently (since a lot of users unfortunately do need `pull_request_target` for 3p triggers).